modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -1052,62 +1079,14 @@ public UserAccount putUserAccountByExternalReferenceCode(
 			String externalReferenceCode, UserAccount userAccount)
 		throws Exception {
 
-		boolean autoPassword = true;
-		String password = userAccount.getPassword();
-
-		if (Validator.isNotNull(password)) {
-			autoPassword = false;
+		User user = _userService.fetchUserByExternalReferenceCode(
```

Comment:
here is fine because in this case we need to do a different thing if is null (we don't throw the exception but we do some operation)